### ISSUE TITLE
Provide mechanism of supplying default values

### ISSUE THREAD

**Issue by timmaffett** [JohnSiegel/parse_json#1]
Your package looks nice - in evaluating if I could use it to replace the mechanism that I currently use to parse json I noticed that there does not seem to be a way of supplying default values for properties that are missing (other than the implicit `null` for things that are marked `optional`).  I do understand that I could mark something as optional and then provide the default in the object's constructor, but I would prefer to place these defaults at the parsing level and leave the constructors with 'required' parameters.

   Something like 
```dart
import 'package:parse_json/parse_json.dart';

final class ExampleObject {
  /// Members with names different from those in the constructor
  final String name;
  final double originalHeight;

  /// Non-final member
  double currentHeight;

  /// Constructor that will be called by `parse`
  const ExampleObject.json({
    required String myString,
    required double myDouble,
  }) : name = myString,
       originalHeight = myDouble,
       currentHeight = myDouble;

  /// Constructor we use for other stuff
  const ExampleObject(this.name, this.originalHeight, this.currentHeight);

  /// Factory function for parsing and constructing from JSON
  factory ExampleObject.fromJson(Map<String, dynamic> json) => 
    parse(
      ExampleObject.json, // the constructor to use
      json,
      {
        'myString': string.optionalWithDefault('circle'),
        'myDouble': float.optionalWithDefault(100.0),
      },
    );
}
```

**Comment by JohnSiegel** [comment]
I hadn't even thought of this. This should be an easy feature to implement. I'll probably have a new update with this feature finished this evening. @timmaffett Thank you for the feedback!

**Comment by JohnSiegel** [comment]
@timmaffett Well, that really didn't take long.. I just published v2.2.0.

This should now work:

```
import 'package:parse_json/parse_json.dart';

final class ExampleObject {
  /// Members with names different from those in the constructor
  final String name;
  final double originalHeight;

  /// Non-final member
  double currentHeight;

  /// Constructor that will be called by `parse`
  const ExampleObject.json({
    required String myString,
    required double myDouble,
  }) : name = myString,
       originalHeight = myDouble,
       currentHeight = myDouble;

  /// Constructor we use for other stuff
  const ExampleObject(this.name, this.originalHeight, this.currentHeight);

  /// Factory function for parsing and constructing from JSON
  factory ExampleObject.fromJson(Map<String, dynamic> json) => 
    parse(
      ExampleObject.json, // the constructor to use
      json,
      {
        'myString': string.withDefault('circle'),
        'myDouble': float.withDefault(100.0),
      },
    );
}
```
